tests/stats/test_stats.py

```diff
@@ -0,0 +1,55 @@
+from utils import interfaces, weblog, features, scenarios, bug
+from utils.tools import logger
+
+"""
+Test scenarios we want:
+    * Generate N spans that will be aggregated together
+        - Must aggregate by:
+            - HTTP status code
+            - peer service tags (todo: can we just rely on the defaults?)
+        - Must have `is_trace_root` on trace root
+        - Must set peer tags
+        - Must have span_kind
+        
+Config:
+- apm_config.peer_tags_aggregation (we should see peer service tags and aggregation by them, note only works on client or producer kind)
+- apm_config.compute_stats_by_span_kind (span_kind will be set and we will calc stats on these spans even when not "top level")
+"""
+
+
+@features.client_side_stats_supported
+class Test_Client_Stats:
+    """Test client-side stats are compatible with Agent implementation"""
+
+    def setup_client_stats(self):
+        for _ in range(5):
+            weblog.get("/stats-unique")
+        for _ in range(3):
+            weblog.get("/stats-unique?code=204")
+
+    # TODO: mark that this is only good for golang net-http for now?
+    # TODO: why are hits and top-level hits separated in python? (and not go)
```

Comment:
Worth clarifying with Python Team / RFC.

This is the kind of discrepancies the system-tests were meant to catch :muscle:. But maybe its just an artifact - it would be good to know for sure.

Reply:
Okay turns out this is an inconsistency in the trace-agent's side of how we aggregate client stats payloads, we're actively rewriting that area of the code so I'm going to put this back into draft until we're sure what the "Correct" behavior going forward should be :P (also unfortunately there is no original "client stats" RFC to refer to here so far as I have been able to find)